.github/workflows/lint.yaml

```diff
@@ -0,0 +1,155 @@
+      - name: Count the number of generic NOLINTs
+        shell: bash
+        run: ./scripts/count_generic_nolints.sh
+  build:
+    name: Build for Lint Tasks
+    runs-on: ubuntu-latest
+    container: debian:stable
+    steps:
+      - name: Update package database
+        run: apt-get update -y
+      # See comments beginning at https://github.com/actions/runner/issues/763#issuecomment-1435474884
+      # Without Git, actions/checkout@v3 will resort to REST and will not
+      # create a .git folder or .git.config. The Problem Matcher looks for
+      # .git/config to find where the root of the repo is, so it must be
+      # present.
+      - name: Install Git
+        run: apt-get -y install git
+      - uses: actions/checkout@v3
+      - name: Install build tools
+        shell: bash
+        run: |
+          apt-get -y install pkg-config libtool autoconf \
+            automake g++ bison flex make bash-completion dh-autoreconf \
+            debhelper devscripts wget python3-pip
+      - name: Install Python lint tools
+        run: python3 -m pip install --no-input cpplint flake8
+      - name: Install build dependencies
+        shell: bash
+        run: |
+          apt-get -y install libcppunit-dev uuid-dev libncurses5-dev \
+            libmicrohttpd-dev protobuf-compiler python3-protobuf \
+            libprotobuf-dev libprotoc-dev zlib1g-dev libftdi-dev \
+            libusb-1.0-0-dev liblo-dev libavahi-client-dev python3-numpy
+      - name: Autoreconf
+        run: autoreconf -i
+      - name: Configure
+        run: ./configure --enable-rdm-tests --enable-ja-rule --enable-e133
+      - name: Build
+        run: make builtfiles VERBOSE=1
+      - name: Archive artifacts to speed up slow GH Actions upload/download
+        shell: bash
+        run: |
+          touch ola-debian-stable-built-source-tree.tar.gz
+          tar --exclude=ola-debian-stable-built-source-tree.tar.gz -cvzf ola-debian-stable-built-source-tree.tar.gz .
+      - name: SHA256 artifact archive
+        run: sha256sum ola-debian-stable-built-source-tree.tar.gz
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ola-debian-stable-built-source-tree
+          path: ola-debian-stable-built-source-tree.tar.gz
+  cpplint:
+    name: cpplint
+    runs-on: ubuntu-latest
+    container: debian:stable
+    needs: build
+    steps:
+      - name: Download built source tree archive
+        uses: actions/download-artifact@v3
+        with:
+          name: ola-debian-stable-built-source-tree
+          path: .
+      - name: SHA256 artifact archive
+        run: sha256sum ola-debian-stable-built-source-tree.tar.gz
+      - name: Unarchive artifacts and delete archive
+        shell: bash
+        run: |
+          tar -xvzf ola-debian-stable-built-source-tree.tar.gz .
+          rm ola-debian-stable-built-source-tree.tar.gz
+      - name: Display structure of extracted files
+        run: ls -alR
```

Comment:
Comment this out? Unless we can leverage the debug option of action builds?

Reply:
Yes, again for debugging.

Reply:
Looks like we could make it configurable via here:
https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging

I'm just wondering if normally it's going to spam the log versus how often we're actually likely to need it?

Reply:
OK, I made the directory listing enabled only with that env var. FWIW, it never bothered me because normally the GitHub UI collapses them. When you download the log archive there is also some segmentation, but I can't remember if it is by job or step.

Reply:
One thing to note to save some headache in the future: It doesn't seem possible to set secrets/env vars in GitHub Actions for branches with open pull requests. Obviously it won't use the target repo secrets/vars because that's a security risk, but it also won't use the source repo vars.

In order to use this debugging feature to fix another issue earlier I had to branch from this branch and enable this variable there.

Reply:
> OK, I made the directory listing enabled only with that env var. FWIW, it never bothered me because normally the GitHub UI collapses them.

Yeah, at least with Travis I often found myself looking at the raw log, I think maybe because it couldn't handle all the lines of the main build or something.

> When you download the log archive there is also some segmentation, but I can't remember if it is by job or step.

That's only by job, but actually the way the build is split, that will help a bit.

> One thing to note to save some headache in the future: It doesn't seem possible to set secrets/env vars in GitHub Actions for branches with open pull requests. Obviously it won't use the target repo secrets/vars because that's a security risk, but it also won't use the source repo vars.

Ah, I must admit I'd just assumed if you made some changes and then looked on your local repo's action run (as a push) that it would give you the debug info.

> In order to use this debugging feature to fix another issue earlier I had to branch from this branch and enable this variable there.

Hmm that's quite a faff.

We can certainly see how we go I think, if we start to find this being really annoying we can certainly revert it and output it all the time.